### DISCUSSION TITLE
Two-factor authentication

### DISCUSSION THREAD

**Discussion by cedricbonhomme** [monarc-project/MonarcAppFO#442] [Ideas]
Add the possibility to authenticate with OTP token.

**Comment by cedricbonhomme** [comment]
Implementation of the issue https://github.com/monarc-project/MonarcAppFO/issues/288

**Comment by cedricbonhomme** [comment]
## 2FA initialization
![Screenshot_20220620_110341](https://user-images.githubusercontent.com/465400/174566914-8d0c721b-d5b1-4bdf-8ef0-55cd55a18ed8.png)

![Screenshot_20220620_110422](https://user-images.githubusercontent.com/465400/174566921-80376409-2972-4f49-a3f1-cbb709edb7b7.png)

## Recovery codes

![Screenshot_20220620_110523](https://user-images.githubusercontent.com/465400/174566923-7b1a34b2-c96d-4490-ae8c-bde9a9e40b83.png)

## Authentication with 2FA

![Screenshot_20220620_110610](https://user-images.githubusercontent.com/465400/174566926-08252622-c7ec-4eb7-8b3f-b959baec6015.png)

**Comment by cedricbonhomme** [comment]
## Global view of user settings

![Screenshot_20220620_111003](https://user-images.githubusercontent.com/465400/174567788-ed280641-0213-4914-9c2b-ad8336c30d76.png)

**Comment by yentl-creyf** [comment]
#446 
#447
